backend/data.go

```diff
@@ -24,10 +25,17 @@ type DataQuery struct {
 	JSON          json.RawMessage
 }
 
-// QueryDataResponse holds the results for a given query.
+// QueryDataResponse contains the results from a QueryDataRequest.
 type QueryDataResponse struct {
-	Frames   []*data.Frame
-	Metadata map[string]string
+	Responses map[string]*DataResponse
+	Metadata  map[string]string
+}
+
+// DataResponse contains the results frim a DataQuery.
+type DataResponse struct {
```

Comment:
QueryResponse?

This is the data for a single query right?

Reply:
That is what it is for. I picked the name to mirror the query: QueryDataRequest contains multiple DataQuery. So QueryDataResponse has multiple DataResponse. So `DataQuery` - `DataResponse`